src/unitxt/standard.py

```diff
@@ -283,52 +275,74 @@ def prepare(self):
 
         self.prepare_refiners()
 
-        self.verblization.steps.append(self.template)
         if self.num_demos > 0:
-            self.verblization.steps.append(
+            self.verbalization.steps.append(
                 AddDemosField(
                     source_stream=self.demos_pool_name,
                     target_field=self.demos_field,
                     sampler=self.sampler,
                 )
             )
-        self.verblization.steps.append(self.system_prompt)
-        self.verblization.steps.append(self.format)
-        if self.augmentor.augment_model_input:
-            self.verblization.steps.append(self.augmentor)
+            if self.templates is not None:
+                self.verbalization.steps.append(
+                    ApplyRandomTemplate(
+                        templates=self.templates, demos_field=self.demos_field
+                    )
+                )
+            else:
+                self.verbalization.steps.append(
+                    ApplySingleTemplate(
+                        template=self.template, demos_field=self.demos_field
+                    )
+                )
+        else:
+            if self.templates is not None:
+                self.verbalization.steps.append(
+                    ApplyRandomTemplate(templates=self.templates)
+                )
+            else:
+                self.verbalization.steps.append(
+                    ApplySingleTemplate(template=self.template)
+                )
 
-        metrics, postprocessors = self.prepare_metrics_and_postprocessors()
+        self.verbalization.steps.append(self.system_prompt)
+        self.verbalization.steps.append(self.format)
+        if self.augmentor.augment_model_input:
+            self.verbalization.steps.append(self.augmentor)
 
-        self.finalize.steps.append(
-            ToUnitxtGroup(
-                group="unitxt",
-                metrics=metrics,
-                postprocessors=postprocessors,
+        if self.postprocessors is not None:
+            self.finalize.steps.append(
+                Set(fields={"postprocessores": self.postprocessors})
             )
-        )
+
+        if self.metrics is not None:
+            self.finalize.steps.append(Set(fields={"metrics": self.metrics}))
+
+        self.finalize.steps.append(Finalize())
 
 
 class StandardRecipeWithIndexes(BaseRecipe):
```

Comment:
It replaced an existing class just was renamed. So far it served us since it's an operator so it is easy to adjust and change its location in the pipeline.

Reply:
I mean why not unite BaseRecipe and standardREcipeWithIndex?  Especially since you added 

   template_card_index: int = NonPositionalField(default=None)
   
  to BaseRecipe

Reply:
Agree and name them all Recipe. In  a seperate PR.